### PR TITLE
(GH-1043) Set-EnvironmentVariable writes an error when insta…

### LINKED ISSUE #1043: Set-EnvironmentVariable writes an error when installing Chocolatey as SYSTEM
### What You Are Seeing?

After installing Chocolatey as SYSTEM, the following handled error is left in $Global:Error:

```
At C:\Windows\TEMP\chocolatey\chocInstall\tools\chocolateyInstall\helpers\functions\Set-EnvironmentVariable.ps1:81 char:5
+     $registryType = $win32RegistryKey.GetValueKind($Name)
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
at Set-EnvironmentVariable, C:\Windows\TEMP\chocolatey\chocInstall\tools\chocolateyInstall\helpers\functions\Set-EnvironmentVariable.ps1: line 81
at Install-ChocolateyEnvironmentVariable, C:\Windows\TEMP\chocolatey\chocInstall\tools\chocolateyInstall\helpers\functions\Install-ChocolateyEnvironmentVariable.ps1: line 100
at Set-ChocolateyInstallFolder, C:\Windows\TEMP\chocolatey\chocInstall\tools\chocolateysetup.psm1: line 166
at Initialize-Chocolatey, C:\Windows\TEMP\chocolatey\chocInstall\tools\chocolateysetup.psm1: line 78
at <ScriptBlock>, C:\Windows\TEMP\chocolatey\chocInstall\tools\chocolateyInstall.ps1: line 16
at <ScriptBlock>, <No file>: line 194
at <ScriptBlock>, E:\Build\WhsInit\Initialize-Computer.ps1: line 183
at <ScriptBlock>, <No file>: line 1
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : IOException
```

### What is Expected?

The code "handles" this exception with a try/catch block with an empty catch block. Unfortunately, PowerShell still puts handled excetpions in $Global:Error. I expect that handled errors should leave no errors in the $Global:Error array. (Our nightly process sends transcripts of its works and we get a report everyday of errors. These errors give use false negatives which lowers the likelihood we'll see real errors.)

### How Did You Get This To Happen? (Steps to Reproduce)

* Open a PowerShell prompt as SYSTEM: `psexec -d -i -s powershell.exe`.
* Delete the ChocolateyInstall environment variable.
* Install chocolatey.
* After the install $Global:Error has the above error in it.

### PR DESCRIPTION
**Pull request by splatteredbits**
Closes #1043

### CODE REVIEW

**Comment by ferventcoder** [comment]
Same here - https://github.com/chocolatey/choco/blob/master/CONTRIBUTING.md#prepare-commits

**Comment by splatteredbits** [comment]
Done.

File: src/chocolatey.resources/helpers/functions/Set-EnvironmentVariable.ps1 [hunk 1]
```diff
@@ -78,7 +78,10 @@ param (
 
   [Microsoft.Win32.RegistryValueKind]$registryType = [Microsoft.Win32.RegistryValueKind]::String
   try {
-    $registryType = $win32RegistryKey.GetValueKind($Name)
+    if( ($win32RegistryKey.GetValueNames() -contains $Name) )
```
Comment (ferventcoder) [inline src/chocolatey.resources/helpers/functions/Set-EnvironmentVariable.ps1]: Out of curiosity, why double parentheses here? Is it personal preference?

**Comment by splatteredbits** [comment]
There are some statements that I use as `if` conditionals that require parenthesis, e.g. `if( (Test-Path -Path $path) )`, so I'm probably in the habit of always doing it. I'm big on consistency.

**Comment by ferventcoder** [comment]
Merged into stable at https://github.com/chocolatey/choco/commit/e7ce79f167f684377b4d9d3c8f97c806de299bad

**Comment by ferventcoder** [comment]
@splatteredbits thanks for your contributions! This will be in 0.10.4.

**Comment by ferventcoder** [comment]
> I'm big on consistency.

@splatteredbits 👍 So are we. Of course as you've likely seen in the PowerShell for Chocolatey, that is six years of learning and finding the right consistency, so apologies if it is/was hard to determine what that consistency should look like.

**Pull request closed without merging**